subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/WithPluginValidation.groovy

```diff
@@ -73,15 +73,23 @@ trait WithPluginValidation {
         }
 
         protected void performValidation(List<String> extraParameters = []) {
+            owner.file("validate-external-gradle-plugin.gradle.kts") << getClass().getResource("validate-external-gradle-plugin.gradle.kts").text
+
             def failsValidation = validations.any { !it.messages.isEmpty() }
-            def validation = owner.runner(["validateExternalPlugins", "--continue", *extraParameters] as String[])
+            def validation = owner.runner([
+                "--init-script", "validate-external-gradle-plugin.gradle.kts",
```

Comment:
❔ Why did you decide to go with the init script? Would it be more difficult to keep it as a plugin?

Reply:
As discussed on Slack current solution is simpler.